### PR TITLE
add stepover command

### PR DESCRIPTION
**Pull request by therealdreg**
## Description

This pull request addresses a limitation in GDB where the "nexti" and "next" commands fail to step over call instructions properly. To resolve this, I have implemented a new feature, "so", specifically designed to enhance debugging efficiency by allowing users to step over call instructions seamlessly.

In my bootcamps, it’s a real pain to deal with this situation, especially for a GDB-newbie.

----

@Ordoviz
```
[Ordoviz](https://github.com/Ordoviz) [1 hour ago](https://github.com/hugsy/gef/pull/1086#discussion_r1571995694)
I also find this very useful for stepping over instructions with rep prefix.
```

--- 
POC

From a x86_64 Linux create a x32 binary 

hello.asm
```
section .data
    msg db 'Hello, World!', 0xA
    len equ $ - msg
    testmsg db 'test!', 0xA
    lentestmsg equ $ - testmsg

section .text
    global _start

_start:
    mov eax, 4
    mov ebx, 1
    mov ecx, msg
    mov edx, len
    int 0x80
    call test

    mov eax, 1
    xor ebx, ebx
    int 0x80

test:
    mov eax, 4
    mov ebx, 1
    mov ecx, testmsg
    mov edx, lentestmsg
    int 0x80
    ret
```

Create executable:
```
nasm -f elf32 hello.asm -o hello.o
ld -m elf_i386 -s -o hello hello.o
strip hello
```

its impossible step over a call in GDB :(

image

The "nexti" and "next" commands do not work as expected :-(

---

**btw, my Python skills are a bit rusty, sorry about that! xD**

## Checklist

-  [X] My code follows the code style of this project.
-  [X] My change includes a change to the documentation, if required.
-  [X] If my change adds new code, [adequate tests](docs/testing.md) have been added.
-  [X] I have read and agree to the **CONTRIBUTING** document.

### CODE REVIEW

File: docs/commands/gnexti.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+## Command `gnexti`
+
+The `gnexti` command allows you to easily step over instructions execution.
```
Comment (Ordoviz) [inline docs/commands/gnexti.md:3]: Please add what this command actually does (continuing to temporary breakpoint at next instruction).

File: gef.py [hunk 2]
```diff
@@ -6157,6 +6166,28 @@ def do_invoke(self, _: List[str], **kwargs: Any) -> None:
         return
 
 
+@register
+class GnextiCommand(GenericCommand):
+    """Step over call instruction"""
```
Comment (Ordoviz) [inline gef.py:6171]: I also find this very useful for stepping over instructions with `rep` prefix.

File: docs/commands/gnexti.md (hunk 1 above)
Comment (therealdreg) [reply docs/commands/gnexti.md]: done!

File: gef.py (hunk 2 above)
Comment (therealdreg) [reply gef.py]: done!

**Comment by therealdreg** [comment]
@hugsy @Ordoviz @Grazfather opinion? looks good for you?

**Comment by hugsy** [comment]
Ok this is wild! This should be a GDB bug, is there any tracking of this?

**Comment by therealdreg** [comment]
> Ok this is wild! This should be a GDB bug, is there any tracking of this?

@hugsy , I don't think it's a bug. I don't believe GDB is designed to step over instructions like call/rep/... for all architectures. Implementing this would be quite a hassle, right? This has been an issue for years.

File: gef.py [hunk 3]
```diff
@@ -6157,6 +6166,28 @@ def do_invoke(self, _: List[str], **kwargs: Any) -> None:
         return
 
 
+@register
+class SoCommand(GenericCommand):
+    """Breaks on the instruction immediately following this one. Ex: Step over call instruction"""
+
+    _cmdline_ = "so"
```
Comment (hugsy) [inline gef.py:6173]: I agree with @Ordoviz `stepover` is more a meaningful name. And make `so` an alias

**Review by hugsy — CHANGES_REQUESTED** [review]
All looks good to me but the name, please change from `so` to `stepover` as a name everywhere, and make `so` an alias 

Thanks!

File: gef.py (hunk 3 above)
Comment (therealdreg) [reply gef.py]: done!

**Comment by therealdreg** [comment]
> All looks good to me but the name, please change from `so` to `stepover` as a name everywhere, and make `so` an alias
> 
> Thanks!

done! @hugsy

**Review by hugsy — APPROVED** [review]
LGTM now

**Pull request merged**